### PR TITLE
RC for Pointless Feedback 4.0

### PR DESCRIPTION
**Pull request by Fosome**
Support Rails 4.x and Ruby 2.x exclusively.

Test matrix includes Ruby 2.0, 2.1, 2.2 and Rails 4.0, 4.1, 4.2.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,7 +32,9 @@ Pointless Feedback has been tested against Rails 3.2+. To install, simply add
 the following to your Gemfile:
 
 ```ruby
-gem 'pointless_feedback', :git => 'git@github.com:vigetlabs/pointless-feedback.git'
+gem 'pointless_feedback',
+  :github => 'vigetlabs/pointless-feedback',
+  :branch => 'rails-4'
```
Comment (reagent) [inline README.md]: Do we ever intend to release this as a gem?

File: app/models/pointless_feedback/message.rb [hunk 2]
```diff
@@ -13,9 +16,11 @@ class Message < ActiveRecord::Base
...
 
     def export_feedback
-      if PointlessFeedback.email_feedback
-        FeedbackMailer.feedback(self).deliver
-      end
+      return unless PointlessFeedback.email_feedback
+
+      # Support Rails < 4.2 and >= 4.2 delivery options
+      mailer = FeedbackMailer.feedback(self)
+      mailer.respond_to?(:deliver_now) ? mailer.deliver_now : mailer.deliver
```
Comment (zporter) [inline app/models/pointless_feedback/message.rb]: `deliver_now` part of the `ActiveJob` thing?

File: lib/pointless_feedback/controllers/helpers.rb [hunk 3]
```diff
@@ -3,7 +3,7 @@ module Controllers
     # Those helpers are convenience methods added to ApplicationController.
     module Helpers
       def after_message_create_path
-        main_app.respond_to?(:root_path) ? main_app.root_path : '/'
+        main_app.try(:root_path) || '/'
```
Comment (zporter) [inline lib/pointless_feedback/controllers/helpers.rb]: :heart:

File: Gemfile [hunk 4]
```diff
@@ -17,6 +17,7 @@ gem "jquery-rails"
 # gem 'debugger'
 
 group :test do
+  gem 'test-unit'
```
Comment (zporter) [inline Gemfile]: I'm more than a little curious as to the reason behind this addition.

File: README.md (hunk 1 above)
Comment (zporter) [inline README.md]: @reagent -- [yes?](https://rubygems.org/gems/pointless_feedback)

**Comment by zporter** [comment]
:+1: though I think the configurable `messages` table name probably should've been a separate branch and PR.

File: README.md (hunk 1 above)
Comment (Fosome) [inline README.md]: Yes.  This is going to be merged and cut (hopefully).  I'll fix this in the docs.

File: app/models/pointless_feedback/message.rb (hunk 2 above)
Comment (Fosome) [inline app/models/pointless_feedback/message.rb]: Yea, `deliver_now` and `deliver_later`.  Sucks the API doesn't support `deliver` as an alias to `deliver_now`.

File: Gemfile (hunk 4 above)
Comment (Fosome) [inline Gemfile]: Ruby 2.x doesn't include test unit anymore.  Or something similar?  http://stackoverflow.com/questions/28252036/how-to-run-existing-test-code-on-ruby-2-2

**Comment by mackermedia** [comment]
:+1:

**Comment by reagent** [comment]
:+1:

**Comment by Fosome** [comment]
Thanks!

**Pull request closed without merging**